services/src/test/scala/cromwell/pipeline/service/ProjectServiceTestImpl.scala

```diff
@@ -0,0 +1,87 @@
+package cromwell.pipeline.service
+import cromwell.pipeline.datastorage.dao.repository.ProjectRepositoryTestImpl
+import cromwell.pipeline.datastorage.dto._
+import cromwell.pipeline.model.wrapper.UserId
+import cromwell.pipeline.service.ProjectService.Exceptions.{ ProjectAccessDeniedException, ProjectNotFoundException }
+import cromwell.pipeline.service.VersioningException.RepositoryException
+
+import scala.concurrent.{ ExecutionContext, Future }
+
+class ProjectServiceTestImpl(testMode: ServiceTestMode, projectRepository: ProjectRepositoryTestImpl)(
+  implicit executionContext: ExecutionContext
+) extends ProjectService {
+
+  private[service] def getUserProjectById(projectId: ProjectId, userId: UserId): Future[Project] =
+    projectRepository.getProjectById(projectId).flatMap(project => getForUserOrFail(project.toSeq, userId))
+
+  def getUserProjectByName(namePattern: String, userId: UserId): Future[Project] = testMode match {
+    case WithException => Future.failed(new RuntimeException("Something went wrong"))
+    case _             => projectRepository.getProjectsByName(namePattern).flatMap(getForUserOrFail(_, userId))
+  }
+
+  private def getForUserOrFail(projects: Seq[Project], userId: UserId): Future[Project] =
+    projects.find(_.ownerId == userId) match {
+      case Some(project)             => Future.successful(project)
+      case None if projects.nonEmpty => Future.failed(new ProjectAccessDeniedException)
+      case _                         => Future.failed(new ProjectNotFoundException)
+    }
+
+  def addProject(request: ProjectAdditionRequest, userId: UserId): Future[Either[VersioningException, Project]] =
+    testMode match {
+      case WithException => Future.successful(Left(RepositoryException("VersioningException.RepositoryException")))
+      case _ =>
+        val testId = "123e4567-e89b-12d3-a456-426614174000"
+        val testVersion = "v1.0.0"
+        val testVersioningId = 42
+        val localProject =
+          LocalProject(
+            projectId = ProjectId(testId),
+            ownerId = userId,
+            name = request.name,
+            active = true,
+            version = PipelineVersion(testVersion)
+          )
+        val newProject = localProject.toProject(RepositoryId(testVersioningId))
+        projectRepository.addProject(newProject).map(_ => Right(newProject))
```

Comment:
Can we just create a dummyProject with name from ProjectAdditionRequest ?
```suggestion
      case _ =>
        val newProject = TestProjectUtils.getDummyProject(
            ownerId = userId,
            name = request.name,
          )
        projectRepository.addProject(newProject).map(_ => Right(newProject))
```

Reply:
Great idea, ill fix it